tests/componenttests/client/tests/base/MediaPipelineTestMethods.cpp

```diff
@@ -409,6 +415,90 @@ void MediaPipelineTestMethods::checkMseVideoSegmentWrittenSecondary(int32_t segm
+
+    // Store where the segment should be written so we can check the data
+    int32_t segmentId = m_audioSegmentCount;
+    writtenAudioSegments.insert({segmentId, *m_locationToWriteAudio[kPrimaryPartition]});
+
+    incrementWriteLocation(kAudioSegments[segmentId].size(), m_locationToWriteAudio[kPrimaryPartition]);
+
+    m_audioSegmentCount++;
+    return segmentId;
+}
+
+void MediaPipelineTestMethods::checkEncryptedAudioSegmentWritten(int32_t segmentId, uint32_t keyIndex)
+{
+    auto it = writtenAudioSegments.find(segmentId);
+    ASSERT_NE(it, writtenAudioSegments.end());
+
+    uint8_t *dataPosition{reinterpret_cast<uint8_t *>(getShmAddress()) + it->second.mediaDataOffset};
+    std::uint32_t *metadataSize{reinterpret_cast<uint32_t *>(dataPosition)};
+    dataPosition += sizeof(uint32_t);
+    MediaSegmentMetadata metadata;
+    ASSERT_TRUE(metadata.ParseFromArray(dataPosition, *metadataSize));
+
+    checkAudioMetadata(metadata, segmentId);
+    checkEncryptionMetadata(metadata, keyIndex);
+    checkHasNoVideoMetadata(metadata);
+    checkHasNoCodacData(metadata);
+    checkHasNoSegmentAlignment(metadata);
+    checkHasNoExtraData(metadata);
+    checkSegmentData(metadata, dataPosition += *metadataSize, kAudioSegments[segmentId]);
+}
+
+int32_t MediaPipelineTestMethods::addSegmentEncryptedVideo(uint32_t keyIndex)
+{
+    EXPECT_LT(m_videoSegmentCount, sizeof(kVideoSegments) / sizeof(kVideoSegments[0]));
+
+    std::unique_ptr<IMediaPipeline::MediaSegment> mseData =
+        std::make_unique<IMediaPipeline::MediaSegmentVideo>(kVideoSourceId, getTimestamp(m_videoSegmentCount),
+                                                            kDuration, kWidth720p, kHeight720p, kFrameRateEmpty);
+    mseData->setData(kVideoSegments[m_videoSegmentCount].size(),
+                     (const uint8_t *)kVideoSegments[m_videoSegmentCount].c_str());
+    addEncryptedDataToSegment(mseData, keyIndex);
+    EXPECT_EQ(m_mediaPipeline->addSegment(m_needDataRequestId, mseData), AddSegmentStatus::OK);
+
+    // Store where the segment should be written so we can check the data
+    int32_t segmentId = m_videoSegmentCount;
+    writtenVideoSegments.insert({segmentId, *m_locationToWriteVideo[kPrimaryPartition]});
+
+    incrementWriteLocation(kVideoSegments[m_videoSegmentCount].size(), m_locationToWriteVideo[kPrimaryPartition]);
+
+    m_videoSegmentCount++;
+    return segmentId;
+}
+
+void MediaPipelineTestMethods::checkEncryptedVideoSegmentWritten(int32_t segmentId, uint32_t keyIndex)
+{
+    auto it = writtenVideoSegments.find(segmentId);
+    ASSERT_NE(it, writtenVideoSegments.end());
+
+    uint8_t *dataPosition{reinterpret_cast<uint8_t *>(getShmAddress()) + it->second.mediaDataOffset};
+    std::uint32_t *metadataSize{reinterpret_cast<uint32_t *>(dataPosition)};
+    dataPosition += sizeof(uint32_t);
+    MediaSegmentMetadata metadata;
+    ASSERT_TRUE(metadata.ParseFromArray(dataPosition, *metadataSize));
```

Comment:
Should lines 445-449 and 487-491 (which are the same) become a function?

Reply:
done